Commit to branch homework4and5.

dmitriopria opened the pull request:
Homework 4
Homework 5
pull request to main.

dmitriopria committed:
Commit to branch homework4and5.

dmitriopria committed:
Names of classes changed to Uppercase first latter

vromantsev commented on src/main/java/Homework_number5.java:
```diff
@@ -0,0 +1,119 @@
+    /* Створити метод findSymbolOccurance. Метод приймає як параметр рядок і символ.
+    Необхідно обчислити, скільки разів символ зустрічається в переданому рядку
+    і повернути це числове значення. */
+    public static int findSymbolOccurance(String inputString, char inputChar) {
```
The input parameters of a method should be validated. For now, would be great if you check if 'inputString' is not null or NullPointerException might be thrown otherwise.

vromantsev commented on src/main/java/Homework_number5.java:
```diff
@@ -0,0 +1,119 @@
+    два рядки (source, target). Необхідно з'ясувати, чи є target (підрядок)
+    частиною рядка source. Якщо так, тоді повернути номер позиції (індекс)
+    першого елемента підрядка у рядку, інакше повернути -1. */
+    public static int findWordPosition(String source, String target) {
```
Please add null validation for both parameters

dmitriopria replied:
done

vromantsev commented on src/main/java/Homework_number5.java:
```diff
@@ -0,0 +1,119 @@
+    /* Створити метод stringReverse. Метод приймає як параметр - рядок.
+    Необхідно її розгорнути та повернути змінений варіант. */
+    public static String reverseString(String inputString) {
+        return "\u202E" + inputString;
```
It is great that you're aware of a Unicode symbol, that allows reading the string from the right to the left =)
It is not a mistake, because the task is perfectly solved by this solution, but would be better for you to practice writing it another way via loops for example. You can leave this version of the solution here, but I would ask you to write another version of it using loops to practice the basic skills in strings and loops.

vromantsev commented on src/main/java/Homework_number5.java:
```diff
@@ -0,0 +1,119 @@
+    /* Створити метод isPalindrome. Метод приймає як параметр - рядок.
+    Необхідно Перевірити, чи є переданий рядок паліндромом.
+    Якщо так, тоді повернуть true, інакше false. */
+    public static boolean isPalindrome(String inputString) {
```
Please add null validation for the inputString.

dmitriopria committed:
If input is null check added to all required methods.
Added one more method of reversing String.
All classes separated to according homework packages.

dmitriopria committed:
Last changes to HW 4 and 5.
